### PR TITLE
test(profiler): use go 1.18.4 in integration test

### PR DESCRIPTION
**Pull request by amchiclet**
Also switching to use gimme (which is well tested and more robust) for the Go installation.

And also printing the go version before running the test, which is useful for debugging when something goes wrong.

### CODE REVIEW

File: profiler/kokoro/integration_test.sh [hunk 1]
```diff
@@ -34,14 +34,29 @@ export GOOGLE_APPLICATION_CREDENTIALS="${KOKORO_KEYSTORE_DIR}/72935_cloud-profil
...
-go install golang.org/dl/go1.17.7
-go1.17.7 download
+# Here we install v1.18.4 which is the current version as of when this code
+# was written.
+mkdir -p /tmp/bin
+GIMME=/tmp/bin/gimme
+retry curl -sL -o "$GIMME" https://raw.githubusercontent.com/travis-ci/gimme/master/gimme
+chmod +x "$GIMME"
+
+export GIMME_GO_VERSION=1.18.4
```
[the remark this answers is not in the document]
Comment (amchiclet) [reply profiler/kokoro/integration_test.sh:44]: Thanks for the feedback! I think you had similar feedback like [last time](https://github.com/googleapis/google-cloud-go/pull/5620#discussion_r808300530).

These tests are extra, on top of the general CI tests that are performed in google-cloud-go.

The Go version being used in kokoro is v1.12. We hope to have a more controlled and custom VM for our tests, but unfortunately for now we will still have to use what kokoro provides.

File: profiler/kokoro/integration_test.sh [hunk 2]
```diff
@@ -34,14 +34,29 @@ export GOOGLE_APPLICATION_CREDENTIALS="${KOKORO_KEYSTORE_DIR}/72935_cloud-profil
...
 # Ensure a newer version of Go is used so it is compatible with newer libraries.
-# Here we install v1.17.7 which is the current version as of when this code
-# was written, following instructions from https://go.dev/doc/manage-install.
-# Go modules might not be on for previous versions of Go, so we also have to
-# enable the module system explicitly.
-export GO111MODULE=on
-go install golang.org/dl/go1.17.7
-go1.17.7 download
+# Here we install v1.18.4 which is the current version as of when this code
+# was written.
```
Comment (wyk9787) [inline profiler/kokoro/integration_test.sh:38]: nit: which is the latest version as of July 2022.

Comment (amchiclet) [reply profiler/kokoro/integration_test.sh]: Done.

**Comment by amchiclet** [comment]
I went ahead an merged this change. If needed, further improvements can be in follow up PRs.

File: profiler/kokoro/integration_test.sh [hunk 3]
```diff
@@ -34,14 +34,28 @@ export GOOGLE_APPLICATION_CREDENTIALS="${KOKORO_KEYSTORE_DIR}/72935_cloud-profil
...
+retry curl -sL -o "$GIMME" https://raw.githubusercontent.com/travis-ci/gimme/master/gimme
+chmod +x "$GIMME"
+
+export GIMME_GO_VERSION=1.18.4
+export GIMME_ENV_PREFIX=/tmp/gimme_envs
+install_go() {
+  "$GIMME"
+  # If gimme was successful, an .env script that sets up proper environment
+  # variables will be created.
+  if [[ -f "${GIMME_ENV_PREFIX}/go${GIMME_GO_VERSION}.env" ]]
```
Comment (aalexand) [inline profiler/kokoro/integration_test.sh:49]: The script uses "set -eo pipefail", so is this conditional really necessary? If the gimme command fails we wouldn't get here. And even if we do, why not just do `source "${GIMME_ENV_PREFIX}/go${GIMME_GO_VERSION}.env"` unconditionally and let the command fail if the file is not there, which should fail the script?

Or is this to enable retries? It is not obvious from the comments.

File: profiler/kokoro/integration_test.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (codyoss) [reply profiler/kokoro/integration_test.sh:43]: @amchiclet I would consider looking into trampoline. You can bootstrap your env with a container. That is how the common CI works.

[the remark this answers is not in the document]
Comment (amchiclet) [reply profiler/kokoro/integration_test.sh:43]: Would love to do things the recommended way. Thanks for the pointer and will look into trampoline when we get a chance.

**Pull request merged**